modules/detection/lidar_driver.py

```diff
@@ -0,0 +1,270 @@
+                )
+                verify_crc = self._create_crc(self._packet_data[0:-2])
+
+                if crc == verify_crc:
+                    return True
+
+        return False
+
+    def _wait_for_packet(self, port, command, timeout=1):
+        """
+        Wait (up to timeout) for a packet of the specified command to be received.
+        """
+        self._packet_parse_state = 0
+        self._packet_data = []
+        self._packet_payload_size = 0
+        self._packet_size = 0
+
+        end_time = time.time() + timeout
+
+        while True:
+            if time.time() >= end_time:
+                return None
+
+            c = port.read(1)
+
+            if len(c) != 0:
+                b = ord(c)
+                if self.parse_packet(b) is True:
+                    if self._packet_data[3] == command:
+                        return self._packet_data
+
+    def _execute_command(self, port, command, write, data=None, timeout=1):
+        """
+        Send a request packet and wait (up to timeout) for a response.
+        """
+        if data is None:
+            data = []
+
+        packet = self._build_packet(command, write, data)
+        retries = 4
+
+        while retries > 0:
+            retries -= 1
+            port.write(packet)
+
+            response = self._wait_for_packet(port, command, timeout)
+
+            if response is not None:
+                return response
+
+        # pylint: disable=broad-exception-raised
+        raise Exception("LWNX command failed to receive a response.")
+        # pylint: enable=broad-exception-raised
+
+    def get_str16_from_packet(self, packet):
+        """
+        Extract a 16 byte string from a string packet.
+        """
+        str16 = ""
+        for i in range(0, 16):
+            if packet[4 + i] == 0:
+                break
+            else:
+                str16 += chr(packet[4 + i])
+
+        return str16
+
+    def print_product_information(self, port):
+        """
+        Prints product information to console.
+        """
+        response = self._execute_command(port, 0, 0, timeout=0.1)
+        print("Product: " + self.get_str16_from_packet(response))
+
+        response = self._execute_command(port, 2, 0, timeout=0.1)
+        print(f"Firmware: {response[6]}.{response[5]}.{response[4]}")
+
+        response = self._execute_command(port, 3, 0, timeout=0.1)
+        print("Serial: " + self.get_str16_from_packet(response))
+
+    def set_update_rate(self, port, value):
+        """
+        Set the frequency of the lidar.
+        Value can be one of:
+        1  = 50 Hz
+        2  = 100 Hz
+        3  = 200 Hz
+        4  = 400 Hz
+        5  = 500 Hz
+        6  = 625 Hz
+        7  = 1000 Hz
+        8  = 1250 Hz
+        9  = 1538 Hz
+        10 = 2000 Hz
+        11 = 2500 Hz
+        12 = 5000 Hz
+        """
+
+        if value < 1 or value > 12:
+            raise ValueError("Invalid update rate value.")
```

Comment:
instead of raising exceptions, return False (raising exceptions is bad because we have to catch them or it takes down the program, which is not a graceful way of handling errors)